ledger/src/blockstore.rs

```diff
@@ -1435,6 +1449,18 @@ impl Blockstore {
             ) {
                 return Err(InsertDataShredError::InvalidShred);
             }
+
+            if let Some(merkle_root_meta) = merkle_root_metas.get(&erasure_set) {
```

Comment:
Under what case would you fail to get merkle_root_meta here, is legacy shred the only possible case?

Reply:
you would fail to get the `merkle_root_meta` if there has been no previous shred from this fec set observed. In case of legacy shred you would still get a `merkle_root_meta` however the `merkle_root` would be `None`.

Reply:
Let's put that into comment then.

Reply:
i think it's inline with the other in memory maps in this function, however if you feel strongly i can add an additional pr to comment all the in memory maps

Reply:
If I just look at this one line I would think both legacy shred and first shred are excluded, I think a comment here would help.